src/app/modules/group/components/manager-permission-dialog/manager-permission-dialog.component.ts

```diff
@@ -74,6 +78,40 @@ export class ManagerPermissionDialogComponent implements OnChanges {
   }
 
   onAccept(): void {
+    if (!this.manager) {
+      throw new Error('Unexpected: Missed manager data');
+    }
+
+    const currentUserId = this.sessionService.session$.value?.user.groupId;
+
+    if (!currentUserId) {
+      throw new Error('Unexpected: Missed current used ID');
+    }
+
+    if (this.manager.id === currentUserId) {
+      const canManageValue = this.form.get('canManage')?.value as GroupManagerPermissionChanges['canManage'];
+
+      if (this.manager.canManage === 'memberships_and_group' && canManageValue !== 'memberships_and_group') {
+        this.confirmationService.confirm({
+          message: $localize`Are you sure to remove from yourself the permission to edit group settings and edit managers ? 
+          You may lose manager access and not be able to restore it.`,
+          header: $localize`Confirm Action`,
+          icon: 'pi pi-exclamation-triangle',
+          acceptLabel: $localize`Yes, save these changes.`,
+          acceptButtonStyleClass: 'p-button-danger',
+          accept: () => {
+            this.update();
+          },
+          rejectLabel: $localize`No`,
+        });
+        return;
+      }
+    }
+
+    this.update();
```

Comment:
To reduce cognitive load, I suggest we handle simplest use cases first and use early returns.
I think it's more readable to handle first the case when manager is not self and return, it would avoid nested `if` statements